packages/components/tree-select/src/tree-select-option.ts

```diff
@@ -0,0 +1,20 @@
+import { defineComponent } from 'vue'
+import ElSelectOption from '@element-plus/components/select/src/option.vue'
+
+const component = defineComponent({
+  extends: ElSelectOption,
+  setup(props, ctx) {
+    const result = (ElSelectOption.setup as NonNullable<any>)(props, ctx)
+
+    delete result.selectOptionClick
```

Comment:
```suggestion
    result.selectOptionClick = null // or undefined.
```
Would this seem better to you?

Reply:
- The `setup` result will overrides `methods` when the property exists.
- Defined in `methods` for better access to `this`.